Automattic/wp-calypso
# PostTypeList: Enlarge and wrap post titles

This PR allows post titles using `PostTypeList` to vary in height and increases their font size.

The initial logic to vary heights by post item was introduced in #17196.  This PR will add another case where post item heights can vary.

At the moment this is not working perfectly:  some multi-line titles that are not _very much_ longer than one line are not correctly measured, even after `requestAnimationFrame`, and the way the variable-height logic is implemented means that post items will change in height after an initial render at the default height.  To get around these issues we may ultimately have to remove `ReactVirtualized` as discussed for example [here](https://github.com/Automattic/wp-calypso/pull/17196#issuecomment-324141510).

### Screenshot

![2017-08-25t17 11 04-0500](https://user-images.githubusercontent.com/227022/29734545-656911e0-89b8-11e7-94a4-0c0a8d742d88.png)

### To test

- Run Calypso with `ENABLE_FEATURES=posts/post-type-list npm start`
- View the list of posts at `/posts` or `/posts/:site`
- Verify that post titles appear using a larger font size and long titles are wrapped across multiple lines on `/posts`, `/pages`, and `/types/:cpt/:site` (e.g. `jetpack-testimonial`).
- Verify that titles re-wrap and post items resize appropriately with changing window size
- Verify that other uses of `PostTypeList` (drafts lists to the right side of `/posts` with and without the `posts/post-type-list` feature flag) are unaffected by this PR.
